device.go

```diff
@@ -0,0 +1,47 @@
+//
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package virtcontainers
+
+import (
+	"os"
+)
+
+// Device that must be available in the container.
+type Device struct {
```

Comment:
I agree with @sboeuf proposal here, I think we should have a `Device` interface and device.go should implement a Device factory based on a minimal set of information it gets about the device we want to attach. We need to know what is the host path of the device, its container path and DAC permissions we want to apply to the container path.
We will get all of this (and more) from an OCI spec.

From device.go, based on the host path, we should infer that we're dealing with a VFIO device, a block device or something else and then create and return the right `Device` interface implementation. From container.go, your `handleDevices()` routine would then only go through all the container devices and call into `Device.attach()`.

Reply:
And I think the `Device` interface should be:

```Go
type Device interface {
      attach()
      detach()
}
```